### DISCUSSION TITLE
Event inside component not working

### DISCUSSION THREAD

**Discussion by britisharmy** [lit/lit#3943] [General]
I have an element with a checkbox inside my lit component. I want to trigger a sweetalert on click.
```

import { LitElement, html, css } from 'https://unpkg.com/lit-element?module';
import Swal from 'sweetalert2';

class NotificationItem extends LitElement {
  static styles = css`
    .notification-item {
      border: 1px solid black;
      border-radius: 10px;
      padding: 20px;
      margin-bottom: 20px;
      transition: opacity 0.3s ease;
    }

    .notification-item:hover {
      opacity: 0.8;
    }

    .custom-checkbox {
      width: 14px;
      height: 14px;
    }
  `;

  static properties = {
    title: { type: String },
    subTitle: { type: String },
    imageUrl: { type: String },
    relativeTime: { type: String },
    dotColor: { type: String },
  };

  handleCheckboxChange(event) {
    if (event.target.checked) {
      Swal.fire('Checkbox Checked!');
    } else {
      Swal.fire('Checkbox Unchecked!');
    }
  }

  render() {
    return html`
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.3.0/css/bootstrap.min.css" />
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css" />

      <div class="notification-item">
        <div class="row align-items-center">
          <div class="col-1">
            <input
              type="checkbox"
              class="custom-checkbox"
              @click=${this.handleCheckboxChange}
            />
            <i class="fas fa-circle" style="color: ${this.dotColor};"></i>
          </div>
          <div class="col-2">
            <img
              src="${this.imageUrl}"
              alt="Profile"
              width="60"
              height="60"
              class="img-fluid rounded-circle"
            />
          </div>
          <div class="col-6">
            <div class="fw-bold">${this.title}</div>
            <div>${this.subTitle}</div>
          </div>
          <div class="col-2">
            <div>${this.relativeTime}</div>
          </div>
          <div class="col-1">
            <i class="fas fa-bookmark"></i>
          </div>
        </div>
      </div>
    `;
  }
}

customElements.define('notification-item', NotificationItem);
```
However, when i display my component

```
<notification-item
  title="An image you viewed has been updated"
  sub-title="Image updated"
 image-url="https://thenewcode.com/assets/images/thumbnails/sarah-parmenter.jpeg"
  relative-time="Posted 2d hours ago"
  dot-color="red"
></notification-item>
```

Checkbox click/change do not trigger opening of sweetalert. Not even an alert or console log shows up when i include that inside function handling click event.

What could be wrong.

**Comment by tbroyer** [comment]
I copy/pasted your code to the Lit Playground, removed Swal as it's not actually an ES module, and replaced it with `alert()` calls, and cannot reproduce, so I'd say it's a problem with your bundler/toolchain.

[Link to the playground](https://lit.dev/playground/#project=W3sibmFtZSI6Im5vdGlmaWNhdGlvbi1pdGVtLmpzIiwiY29udGVudCI6IlxuaW1wb3J0IHsgTGl0RWxlbWVudCwgaHRtbCwgY3NzIH0gZnJvbSAnbGl0JztcblxuY2xhc3MgTm90aWZpY2F0aW9uSXRlbSBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgIC5ub3RpZmljYXRpb24taXRlbSB7XG4gICAgICBib3JkZXI6IDFweCBzb2xpZCBibGFjaztcbiAgICAgIGJvcmRlci1yYWRpdXM6IDEwcHg7XG4gICAgICBwYWRkaW5nOiAyMHB4O1xuICAgICAgbWFyZ2luLWJvdHRvbTogMjBweDtcbiAgICAgIHRyYW5zaXRpb246IG9wYWNpdHkgMC4zcyBlYXNlO1xuICAgIH1cblxuICAgIC5ub3RpZmljYXRpb24taXRlbTpob3ZlciB7XG4gICAgICBvcGFjaXR5OiAwLjg7XG4gICAgfVxuXG4gICAgLmN1c3RvbS1jaGVja2JveCB7XG4gICAgICB3aWR0aDogMTRweDtcbiAgICAgIGhlaWdodDogMTRweDtcbiAgICB9XG4gIGA7XG5cbiAgc3RhdGljIHByb3BlcnRpZXMgPSB7XG4gICAgdGl0bGU6IHsgdHlwZTogU3RyaW5nIH0sXG4gICAgc3ViVGl0bGU6IHsgdHlwZTogU3RyaW5nIH0sXG4gICAgaW1hZ2VVcmw6IHsgdHlwZTogU3RyaW5nIH0sXG4gICAgcmVsYXRpdmVUaW1lOiB7IHR5cGU6IFN0cmluZyB9LFxuICAgIGRvdENvbG9yOiB7IHR5cGU6IFN0cmluZyB9LFxuICB9O1xuXG4gIGhhbmRsZUNoZWNrYm94Q2hhbmdlKGV2ZW50KSB7XG4gICAgaWYgKGV2ZW50LnRhcmdldC5jaGVja2VkKSB7XG4gICAgICBhbGVydCgnQ2hlY2tib3ggQ2hlY2tlZCEnKTtcbiAgICB9IGVsc2Uge1xuICAgICAgYWxlcnQoJ0NoZWNrYm94IFVuY2hlY2tlZCEnKTtcbiAgICB9XG4gIH1cblxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgXG4gICAgICA8bGluayByZWw9XCJzdHlsZXNoZWV0XCIgaHJlZj1cImh0dHBzOi8vY2RuanMuY2xvdWRmbGFyZS5jb20vYWpheC9saWJzL2Jvb3RzdHJhcC81LjMuMC9jc3MvYm9vdHN0cmFwLm1pbi5jc3NcIiAvPlxuICAgICAgPGxpbmsgcmVsPVwic3R5bGVzaGVldFwiIGhyZWY9XCJodHRwczovL2NkbmpzLmNsb3VkZmxhcmUuY29tL2FqYXgvbGlicy9mb250LWF3ZXNvbWUvNS4xNS4zL2Nzcy9hbGwubWluLmNzc1wiIC8-XG5cbiAgICAgIDxkaXYgY2xhc3M9XCJub3RpZmljYXRpb24taXRlbVwiPlxuICAgICAgICA8ZGl2IGNsYXNzPVwicm93IGFsaWduLWl0ZW1zLWNlbnRlclwiPlxuICAgICAgICAgIDxkaXYgY2xhc3M9XCJjb2wtMVwiPlxuICAgICAgICAgICAgPGlucHV0XG4gICAgICAgICAgICAgIHR5cGU9XCJjaGVja2JveFwiXG4gICAgICAgICAgICAgIGNsYXNzPVwiY3VzdG9tLWNoZWNrYm94XCJcbiAgICAgICAgICAgICAgQGNsaWNrPSR7dGhpcy5oYW5kbGVDaGVja2JveENoYW5nZX1cbiAgICAgICAgICAgIC8-XG4gICAgICAgICAgICA8aSBjbGFzcz1cImZhcyBmYS1jaXJjbGVcIiBzdHlsZT1cImNvbG9yOiAke3RoaXMuZG90Q29sb3J9O1wiPjwvaT5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgICA8ZGl2IGNsYXNzPVwiY29sLTJcIj5cbiAgICAgICAgICAgIDxpbWdcbiAgICAgICAgICAgICAgc3JjPVwiJHt0aGlzLmltYWdlVXJsfVwiXG4gICAgICAgICAgICAgIGFsdD1cIlByb2ZpbGVcIlxuICAgICAgICAgICAgICB3aWR0aD1cIjYwXCJcbiAgICAgICAgICAgICAgaGVpZ2h0PVwiNjBcIlxuICAgICAgICAgICAgICBjbGFzcz1cImltZy1mbHVpZCByb3VuZGVkLWNpcmNsZVwiXG4gICAgICAgICAgICAvPlxuICAgICAgICAgIDwvZGl2PlxuICAgICAgICAgIDxkaXYgY2xhc3M9XCJjb2wtNlwiPlxuICAgICAgICAgICAgPGRpdiBjbGFzcz1cImZ3LWJvbGRcIj4ke3RoaXMudGl0bGV9PC9kaXY-XG4gICAgICAgICAgICA8ZGl2PiR7dGhpcy5zdWJUaXRsZX08L2Rpdj5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgICA8ZGl2IGNsYXNzPVwiY29sLTJcIj5cbiAgICAgICAgICAgIDxkaXY-JHt0aGlzLnJlbGF0aXZlVGltZX08L2Rpdj5cbiAgICAgICAgICA8L2Rpdj5cbiAgICAgICAgICA8ZGl2IGNsYXNzPVwiY29sLTFcIj5cbiAgICAgICAgICAgIDxpIGNsYXNzPVwiZmFzIGZhLWJvb2ttYXJrXCI-PC9pPlxuICAgICAgICAgIDwvZGl2PlxuICAgICAgICA8L2Rpdj5cbiAgICAgIDwvZGl2PlxuICAgIGA7XG4gIH1cbn1cblxuY3VzdG9tRWxlbWVudHMuZGVmaW5lKCdub3RpZmljYXRpb24taXRlbScsIE5vdGlmaWNhdGlvbkl0ZW0pOyJ9LHsibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFET0NUWVBFIGh0bWw-XG48aGVhZD5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9ub3RpZmljYXRpb24taXRlbS5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxub3RpZmljYXRpb24taXRlbVxuICB0aXRsZT1cIkFuIGltYWdlIHlvdSB2aWV3ZWQgaGFzIGJlZW4gdXBkYXRlZFwiXG4gIHN1Yi10aXRsZT1cIkltYWdlIHVwZGF0ZWRcIlxuIGltYWdlLXVybD1cImh0dHBzOi8vdGhlbmV3Y29kZS5jb20vYXNzZXRzL2ltYWdlcy90aHVtYm5haWxzL3NhcmFoLXBhcm1lbnRlci5qcGVnXCJcbiAgcmVsYXRpdmUtdGltZT1cIlBvc3RlZCAyZCBob3VycyBhZ29cIlxuICBkb3QtY29sb3I9XCJyZWRcIlxuPjwvbm90aWZpY2F0aW9uLWl0ZW0-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d)

**Reply by britisharmy** [reply]
You are right. My browser wasn't catching the ES module as an error. I am loading every js file via cdn so this is wrong `import Swal from 'sweetalert2';`

**Comment by britisharmy** [comment]
I fixed it this way
```

import { LitElement, html, css } from 'https://unpkg.com/lit-element?module';

class NotificationItem extends LitElement {
  static styles = css`
    .notification-item {
      border: 1px solid black;
      border-radius: 10px;
      padding: 20px;
      margin-bottom: 20px;
      transition: opacity 0.3s ease;
    }

    .notification-item:hover {
      opacity: 0.8;
    }

    .custom-checkbox {
      width: 14px;
      height: 14px;
    }
  `;

  static properties = {
    title: { type: String },
    subtitle: { type: String },
    imageurl: { type: String },
    relativetime: { type: String },
    dotcolor: { type: String },
  };

  handleCheckboxClick() {
    // Open SweetAlert here
    Swal.fire('Checkbox Clicked!');
  }

  render() {
    return html`
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.3.0/css/bootstrap.min.css" />
      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css" />

      <div class="notification-item">
        <div class="row align-items-center">
          <div class="col-1">
            <input
              type="checkbox"
              class="custom-checkbox"
              @click=${this.handleCheckboxClick}
            />
            <i class="fas fa-circle" style="color: ${this.dotcolor};"></i>
          </div>
          <div class="col-2">
            <img
              src="${this.imageurl}"
              alt="Profile"
              width="60"
              height="60"
              class="img-fluid rounded-circle"
            />
          </div>
          <div class="col-6">
            <div class="fw-bold">${this.title}</div>
            <div>${this.subtitle}</div>
          </div>
          <div class="col-2">
            <div>${this.relativetime}</div>
          </div>
          <div class="col-1">
            <i class="fas fa-bookmark"></i>
          </div>
        </div>
      </div>
    `;
  }
}

customElements.define('notification-item', NotificationItem);

```

and this is how i am using it

```
    <notification-item
  title="An image you viewed has been updatedsss"
  subtitle="Imagedddd updated"
 imageurl="https://images.freeimages.com/fic/images/icons/747/network/256/user_group.png"
  relativetime="Posted 2d hours ago"
  dotcolor="green"
></notification-item>
```

and it works
